[TST] Compat with macos and window, FailedHealthCheck

This PR #1143 is almost done.
But it still left one problem.
HealthCheck always failed on macos and window system like following.

https://github.com/pyjanitor-devs/pyjanitor/actions/runs/3281183133/jobs/5402869041#step:5:2147

```python
_____________________________ test_df_key_hashable _____________________________
[gw2] darwin -- Python 3.10.6 /Users/runner/miniconda3/envs/test/bin/python

    @given(df=df_strategy())
>   def test_df_key_hashable(df):
E   hypothesis.errors.FailedHealthCheck: Data generation is extremely slow: Only produced 6 valid examples in 1.08 seconds (0 invalid ones and 4 exceeded maximum size). Try decreasing size of the data you're generating (with e.g. max_size or max_leaves parameters).
E   See https://hypothesis.readthedocs.io/en/latest/healthchecks.html for more information about this. If you want to disable just this health check, add HealthCheck.too_slow to the suppress_health_check settings for this test.

tests/functions/test_expand_grid.py:40: FailedHealthCheck
```

I thought wether we could use a small size date like `max_size=10` or a small one?

https://github.com/pyjanitor-devs/pyjanitor/blob/f0c79066c3faa0762626339054692401a7b8db58/janitor/testing_utils/strategies.py#L21-L43

--- a/tests/functions/test_case_when.py
+++ b/tests/functions/test_case_when.py
@@ -201,6 +201,7 @@ def test_case_when_default_list_like(df):
 
 
 @given(df=categoricaldf_strategy())
+@settings(deadline=None)
 def test_case_when_default_index(df):
     """
     Test case_when for scenarios where `default` is an index.

--- a/tests/functions/test_case_when.py
+++ b/tests/functions/test_case_when.py
@@ -184,6 +184,7 @@ def test_case_when_default_array(df):
 
 
 @given(df=categoricaldf_strategy())
+@settings(deadline=None)
 def test_case_when_default_list_like(df):
     """
     Test case_when for scenarios where `default` is list-like,

--- a/tests/functions/test_case_when.py
+++ b/tests/functions/test_case_when.py
@@ -166,6 +166,7 @@ def test_case_when_replacement_callable(df):
 
 
 @given(df=categoricaldf_strategy())
+@settings(deadline=None)
 def test_case_when_default_array(df):
     """
     Test case_when for scenarios where `default` is array-like

--- a/tests/functions/test_encode_categorical.py
+++ b/tests/functions/test_encode_categorical.py
@@ -234,6 +234,7 @@ def test_empty_col_sort(df):
 
 @pytest.mark.functions
 @given(df=df_strategy())
+@settings(deadline=None)
 def test_empty_col_appearance(df):
     """
     Raise ValueError if a string is provided,

--- a/tests/functions/test_encode_categorical.py
+++ b/tests/functions/test_encode_categorical.py
@@ -271,6 +271,7 @@ def test_all_None(df):
 
 @pytest.mark.functions
 @given(df=categoricaldf_strategy())
+@settings(deadline=None)
 def test_all_cat_None_1(df):
     """
     Test output where a string is provided.

--- a/tests/functions/test_encode_categorical.py
+++ b/tests/functions/test_encode_categorical.py
@@ -258,6 +258,7 @@ def test_empty_col_appearance(df):
 
 @pytest.mark.functions
 @given(df=categoricaldf_strategy())
+@settings(deadline=None)
 def test_all_None(df):
     """
     Test output where value is None.

--- a/tests/functions/test_expand_grid.py
+++ b/tests/functions/test_expand_grid.py
@@ -13,6 +13,7 @@ from janitor.testing_utils.strategies import categoricaldf_strategy
 
 
 @given(df=df_strategy())
+@settings(deadline=None)
 def test_others_not_dict(df):
     """Raise Error if `others` is not a dictionary."""
     with pytest.raises(TypeError):

--- a/tests/functions/test_expand_grid.py
+++ b/tests/functions/test_expand_grid.py
@@ -33,6 +33,7 @@ def test_others_empty():
 
 
 @given(df=df_strategy())
+@settings(deadline=None)
 def test_df_key(df):
     """Raise error if df exists and df_key is not supplied."""
     with pytest.raises(KeyError):

--- a/tests/math/test_ecdf.py
+++ b/tests/math/test_ecdf.py
@@ -1,10 +1,12 @@
 import numpy as np
 import pytest
 from hypothesis import given
+from hypothesis import settings
 from hypothesis.extra.pandas import series
 
 
 @given(s=series(dtype=np.number))
+@settings(deadline=None)
 def test_ecdf(s):
     """A simple execution test."""
     if s.isna().sum() > 0:

